-1

Tested at: a34ab5cb44a35f8aada847cc14096a2c91f34891


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/99423efc0bd3b53dba1ac570b2b752285f471262
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18287/19053/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18287/19053/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18287/19053/summary.html

I found follow errors while testing this PR

**Failed tests:  ClangBuild**

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 16 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-10-1100/src/EventFilter/Utilities/plugins/DaqFakeReader.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-10-1100/src/EventFilter/Utilities/plugins/TcdsRawToDigi.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-10-1100/src/EventFilter/Utilities/plugins/RawEventFileWriterForBU.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-10-1100/src/EventFilter/Utilities/plugins/RecoEventWriterForFU.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-10-1100/src/EventFilter/Utilities/plugins/SealModule.cc 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-10-1100/src/EventFilter/Utilities/src/json_value.cpp:100:39: error: default initialization of an object of const type 'const Json::DefaultValueAllocator' without a user-provided default constructor
   static const DefaultValueAllocator defaultAllocator;
                                      ^
                                                      {}
1 error generated.
gmake: *** [tmp/slc6_amd64_gcc530/src/EventFilter/Utilities/src/EventFilterUtilities/json_value.o] Error 1
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/99423efc0bd3b53dba1ac570b2b752285f471262
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18287/19053/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18287/19053/git-merge-result

--- a/EventFilter/Utilities/interface/EvFDaqDirector.h
+++ b/EventFilter/Utilities/interface/EvFDaqDirector.h
@@ -29,8 +29,8 @@ class SystemBounds;
 class GlobalContext;
 class StreamID;
 
-class InputFile;
-class InputChunk;
+struct InputFile;
+struct InputChunk;
 
 namespace edm {
   class PathsAndConsumesOfModulesBase;

--- a/EventFilter/Utilities/interface/FedRawDataInputSource.h
+++ b/EventFilter/Utilities/interface/FedRawDataInputSource.h
@@ -28,8 +28,8 @@ class FEDRawDataCollection;
 class InputSourceDescription;
 class ParameterSet;
 
-class InputFile;
-class InputChunk;
+struct InputFile;
+struct  InputChunk;
 
 namespace evf {
 class FastMonitoringService;
@@ -42,8 +42,8 @@ class DataPointDefinition;
 
 class FedRawDataInputSource: public edm::RawInputSource {
 
-friend class InputFile;
-friend class InputChunk;
+friend struct InputFile;
+friend struct InputChunk;
 
 public:
   explicit FedRawDataInputSource(edm::ParameterSet const&,edm::InputSourceDescription const&);

--- a/EventFilter/Utilities/plugins/ExceptionGenerator.cc
+++ b/EventFilter/Utilities/plugins/ExceptionGenerator.cc
@@ -137,7 +137,7 @@ namespace evf{
     timingHisto_->SetEntries(24934);
   }
 
-  void ExceptionGenerator::beginRun(edm::Run& r, const edm::EventSetup& iSetup)
+  void ExceptionGenerator::beginRun(const edm::Run& r, const edm::EventSetup& iSetup)
   {
 
     gettimeofday(&tv_start_,0);

--- a/EventFilter/Utilities/plugins/ExceptionGenerator.h
+++ b/EventFilter/Utilities/plugins/ExceptionGenerator.h
@@ -19,9 +19,9 @@ namespace evf{
 						   
       explicit ExceptionGenerator( const edm::ParameterSet&);
       ~ExceptionGenerator(){};
-      void beginRun(edm::Run& r, const edm::EventSetup& iSetup);
-      void analyze(const edm::Event & e, const edm::EventSetup& c);
-      void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&);
+      void beginRun(const edm::Run& r, const edm::EventSetup& iSetup) override;
+      void analyze(const edm::Event & e, const edm::EventSetup& c) override;
+      void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
 
     private:
       int actionId_;

--- a/EventFilter/Utilities/src/json_value.cpp
+++ b/EventFilter/Utilities/src/json_value.cpp
@@ -59,6 +59,8 @@ ValueAllocator::~ValueAllocator()
 class DefaultValueAllocator : public ValueAllocator
 {
 public:
+   DefaultValueAllocator() = default;
+
    virtual ~DefaultValueAllocator()
    {
    }

--- a/EventFilter/Utilities/src/json_value.cpp
+++ b/EventFilter/Utilities/src/json_value.cpp
@@ -59,7 +59,7 @@ ValueAllocator::~ValueAllocator()
 class DefaultValueAllocator : public ValueAllocator
 {
 public:
-   DefaultValueAllocator() = default;
+   DefaultValueAllocator() {}
 
    virtual ~DefaultValueAllocator()
    {

